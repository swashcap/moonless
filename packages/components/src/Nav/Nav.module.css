/** Nav */

/** Nav: List */

.NavList {
  display: flex;
  list-style: none;
  margin: 0;
  padding: 0;
}

/** Nav: Link */

.NavLink {
  background: var(--component-nav-link-background-color-base);
  color: var(--component-nav-link-text-color-base);
  display: inline-block;
  padding: var(--component-nav-link-padding);
  text-decoration: none;
}

.NavLink:hover {
  background: var(--component-nav-link-background-color-hover);
  color: var(--component-nav-link-text-color-hover);
}

.NavLink:focus,
.NavLink:active {
  background: var(--component-nav-link-background-color-active);
  color: var(--component-nav-link-text-color-active);
}

/** Nav: Link: Current */

.NavLinkCurrent {
  background: var(--color-theme-primary-background-color-base);
  color: var(--color-theme-primary-text-color-base);
}

.NavLinkCurrent:hover {
  background: var(--color-theme-primary-background-color-hover);
  color: var(--color-theme-primary-text-color-hover);
}

.NavLinkCurrent:focus,
.NavLinkCurrent:active {
  background: var(--color-theme-primary-background-color-active);
  color: var(--color-theme-primary-text-color-active);
}
