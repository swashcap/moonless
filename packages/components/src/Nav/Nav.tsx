import { Component } from 'react';
import { clsx } from 'moonless-utils';

import styles from './Nav.module.css';

export type NavProps = React.DetailedHTMLProps<
  React.HTMLAttributes<HTMLDivElement>,
  HTMLDivElement
>;

export interface NavItemProps
  extends React.DetailedHTMLProps<
    React.AnchorHTMLAttributes<HTMLAnchorElement>,
    HTMLAnchorElement
  > {
  isCurrent?: boolean;
}

export class Nav extends Component<NavProps> {
  static Item = ({ children, className, isCurrent, ...rest }: NavItemProps) => {
    return (
      <li className={className}>
        <a
          className={clsx(styles.NavLink, isCurrent && styles.NavLinkCurrent)}
          {...rest}
        >
          {children}
        </a>
      </li>
    );
  };

  render() {
    const { children, ...rest } = this.props;

    return (
      <nav {...rest}>
        <ul className={styles.NavList}>{children}</ul>
      </nav>
    );
  }
}
