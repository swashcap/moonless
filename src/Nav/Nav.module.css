/** Nav */

.Nav {
  background: var(--color-blue);
  color: var(--color-white);
}

/** Nav: List */

.NavList {
  display: flex;
  list-style: none;
  margin: 0;
  padding: 0;
}

/** Nav: Link */

.NavLink {
  color: inherit;
  display: inline-block;
  padding: var(--space-100);
  text-decoration: none;
}
