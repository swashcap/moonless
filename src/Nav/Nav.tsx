import React from 'react';

import { clsx } from '../utils/clsx';
import styles from './Nav.module.css';

export type NavProps = React.DetailedHTMLProps<
  React.HTMLAttributes<HTMLDivElement>,
  HTMLDivElement
>;

export type NavItemProps = React.DetailedHTMLProps<
  React.AnchorHTMLAttributes<HTMLAnchorElement>,
  HTMLAnchorElement
>;

export class Nav extends React.Component<NavProps> {
  static Item = ({ children, className, ...rest }: NavItemProps) => {
    return (
      <li className={className}>
        <a className={styles.NavLink} {...rest}>
          {children}
        </a>
      </li>
    );
  };

  render() {
    const { children, className, ...rest } = this.props;

    return (
      <nav className={clsx(styles.Nav, className)} {...rest}>
        <ul className={styles.NavList}>{children}</ul>
      </nav>
    );
  }
}
