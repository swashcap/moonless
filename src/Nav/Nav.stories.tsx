import React from 'react';

import { Nav } from './Nav';

export default {
  component: Nav,
  title: 'Component/Nav',
};

export const Default = () => <Nav />;

export const WithItems = () => (
  <Nav>
    <Nav.Item href="#/">Home</Nav.Item>
    <Nav.Item href="#/page-2/">Page 2</Nav.Item>
    <Nav.Item href="#/page-3/" isCurrent>
      Page 3
    </Nav.Item>
  </Nav>
);
