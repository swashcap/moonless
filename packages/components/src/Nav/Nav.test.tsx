import { render } from '@testing-library/react';

import { Nav } from './Nav';

describe('Nav', () => {
  test('It should render.', () => {
    const { container } = render(<Nav />);

    expect(container).toMatchSnapshot();
  });
});
