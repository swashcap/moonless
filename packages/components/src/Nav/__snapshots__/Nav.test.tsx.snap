// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav It should render. 1`] = `
<div>
  <nav>
    <ul
      class="NavList"
    />
  </nav>
</div>
`;
